src/stm32l0.c

```diff
@@ -0,0 +1,1076 @@
+    gdb_outf ("0x%08x: 0x%04x 0x%04x %s\n",
+              addr, val & 0xffff, (val >> 16) & 0xffff,
+              ((val & 0xffff) == ((~val >> 16) & 0xffff)) ? "OK" : "ERR");
+  }
+
+  if (stm32lx_is_stm32l1 (t)) {
+    uint32_t optr   = adiv5_ap_mem_read (ap, STM32Lx_NVM_OPTR(nvm));
+    uint8_t  rdprot = (optr >> STM32L1_NVM_OPTR_RDPROT_S)
+      & STM32L1_NVM_OPTR_RDPROT_M;
+    if (rdprot == STM32L1_NVM_OPTR_RDPROT_0)
+      rdprot = 0;
+    else if (rdprot == STM32L1_NVM_OPTR_RDPROT_2)
+      rdprot = 2;
+    else
+      rdprot = 1;
+    gdb_outf ("OPTR: 0x%08x, RDPRT %d, SPRMD %d, "
+              "BOR %d, WDG_SW %d, nRST_STP %d, nRST_STBY %d, nBFB2 %d\n",
+              optr, rdprot,
+              (optr &  STM32L1_NVM_OPTR_SPRMOD)     ? 1 : 0,
+              (optr >> STM32L1_NVM_OPTR_BOR_LEV_S) & STM32L1_NVM_OPTR_BOR_LEV_M,
+              (optr &  STM32L1_NVM_OPTR_WDG_SW)     ? 1 : 0,
+              (optr &  STM32L1_NVM_OPTR_nRST_STOP)  ? 1 : 0,
+              (optr &  STM32L1_NVM_OPTR_nRST_STDBY) ? 1 : 0,
+              (optr &  STM32L1_NVM_OPTR_nBFB2)      ? 1 : 0);
+  }
+  else {
+    uint32_t optr   = adiv5_ap_mem_read (ap, STM32Lx_NVM_OPTR(nvm));
+    uint8_t  rdprot = (optr >> STM32L0_NVM_OPTR_RDPROT_S)
+      & STM32L0_NVM_OPTR_RDPROT_M;
+    if (rdprot == STM32L0_NVM_OPTR_RDPROT_0)
+      rdprot = 0;
+    else if (rdprot == STM32L0_NVM_OPTR_RDPROT_2)
+      rdprot = 2;
+    else
+      rdprot = 1;
+    gdb_outf ("OPTR: 0x%08x, RDPROT %d, WPRMOD %d, WDG_SW %d, BOOT1 %d\n",
+              optr, rdprot,
+              (optr & STM32L0_NVM_OPTR_WPRMOD) ? 1 : 0,
+              (optr & STM32L0_NVM_OPTR_WDG_SW) ? 1 : 0,
+              (optr & STM32L0_NVM_OPTR_BOOT1)  ? 1 : 0);
+  }
+
+  goto done;
+
+ usage:
+  gdb_out ("usage: monitor option [ARGS]\n");
+  gdb_out ("  show                   - Show options in NVM and as loaded\n");
+  gdb_out ("  obl_launch             - Reload options from NVM\n");
+  gdb_out ("  write <addr> <value16> - Set option half-word; "
+           "complement computed\n");
+  gdb_out ("  raw <addr> <value32>   - Set option word\n");
+  gdb_outf ("The value of <addr> must be word aligned and from 0x%08x "
+            "to +0x%x\n",
+            STM32Lx_NVM_OPT_PHYS,
+            STM32Lx_NVM_OPT_PHYS + opt_size - sizeof (uint32_t));
+
+ done:
+  stm32lx_nvm_lock (ap, nvm);
+  return true;
+}
+
+
+static bool stm32lx_cmd_eeprom (target* t, int argc, char** argv)
+{
+  ADIv5_AP_t*    ap  = adiv5_target_ap (t);
+  const uint32_t nvm = stm32lx_nvm_phys (t);
+
+  if (!stm32lx_nvm_prog_data_unlock (ap, nvm)) {
+    gdb_out ("unable to unlock EEPROM\n");
+    return true;
+  }
+
+  size_t cb = strlen (argv[1]);
+
+  if (argc == 4) {
+    uint32_t addr = strtoul (argv[2], NULL, 0);
+    uint32_t val  = strtoul (argv[3], NULL, 0);
+
+    if (   addr <  STM32Lx_NVM_EEPROM_PHYS
+        || addr >= STM32Lx_NVM_EEPROM_PHYS + stm32lx_nvm_eeprom_size (t))
+      goto usage;
+
+#if 0
```

Comment:
I'll see if I can fix this instead of removing it.  I think the catch is that the eeprom write code presently doesn't permit byte writes.  IIRC, the MCU does so this is really a shortcoming of my implementation.